jdaviz/app.py

```diff
@@ -1816,18 +1824,39 @@ def _reparent_subsets(self, old_parent, new_parent=None):
                                     dummy_x = old_xc + r
                                     x2, y2 = pixel_to_pixel(old_parent.coords, new_parent.coords,
                                                             dummy_x, old_yc)
-                                    new_radius = np.abs(x2 - x)
+                                    # Need to use x and y in this radius calculation because the
+                                    # new orientation is likely rotated compared to the original.
+                                    new_radius = np.sqrt((x2 - x)**2 + (y2 - y)**2)
                                     setattr(roi, att, new_radius)
 
                         elif type(roi) is RectangularROI:
-                            x_min, y_min = pixel_to_pixel(old_parent.coords, new_parent.coords,
-                                                          roi.xmin, roi.ymin)
-                            x_max, y_max = pixel_to_pixel(old_parent.coords, new_parent.coords,
-                                                          roi.xmax, roi.ymax)
-                            roi.xmin = x_min
-                            roi.xmax = x_max
-                            roi.ymin = y_min
-                            roi.ymax = y_max
+                            x1, y1 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                    roi.xmin, roi.ymin)
+                            x2, y2 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                    roi.xmin, roi.ymax)
+                            x3, y3 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                    roi.xmax, roi.ymax)
+                            x3, y3 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                    roi.xmax, roi.ymin)
```

Comment:
`pixel_to_pixel` can vectorize, I think, see https://docs.astropy.org/en/stable/api/astropy.wcs.utils.pixel_to_pixel.html , so you really only need one call to that function.

Reply:
Hmm, I tried calling it a couple different ways, and it doesn't seem like vectorizing analogously to `pixel_to_world` works for `pixel_to_pixel`. I did remove an extra call, realized I was setting then overwriting x3 and y3.

Reply:
Oh, OK, maybe the API doc is misleading then. Thanks for trying!